x/campaign/keeper/msg_server_update_total_supply.go

```diff
@@ -35,6 +35,12 @@ func (k msgServer) UpdateTotalSupply(goCtx context.Context, msg *types.MsgUpdate
 		return nil, sdkerrors.Wrapf(types.ErrMainnetInitialized, "%d", msg.CampaignID)
 	}
 
+	// Validate provided totalSupply
+	totalSupplyRange := k.TotalSupplyRange(ctx)
+	if err := types.ValidateTotalSupply(msg.TotalSupplyUpdate, totalSupplyRange); err != nil {
+		return nil, sdkerrors.Wrap(types.ErrInvalidTotalSupply, err.Error())
+	}
```

Comment:
We must introduce a test case in message test for this condition

Reply:
right, I was thinking about it yesterday when I logged off, completely forgot to add test cases for this check in both messages. Will do